## Punto 1

In [1]:
from datetime import datetime, timedelta
import math
import numpy as np
import matplotlib.pyplot as plt #para graficar

#CONSTANTES:

# se supone que el asteroide va alrededor del sol
K = 0.01720209908 #U.A.^(3/2)/d

a = 2.6785123 # U.A
e = 0.2543422


In [2]:
T = (2*math.pi*a**(3/2))/K  #d : días
print("El periodo orbital es T = ", T, "días")

El periodo orbital es T =  1601.1753337699245 días


In [3]:
n = 360/T # °/d
print("El movimiento medio es n = ", n, "°/d")

El movimiento medio es n =  0.22483484001242365 °/d


In [4]:
def days_between(utc_time1, utc_time2): #Nos retorna la diferencia en días con fracción decimal
    
  time1 = datetime.strptime(utc_time1, "%Y-%m-%d %H:%M:%S")
  time2 = datetime.strptime(utc_time2, "%Y-%m-%d %H:%M:%S")
  delta = time2 - time1
  return delta.total_seconds() / (24 * 3600) #how many days with decimals

utc_time1 = "2005-05-05 00:00:00" #tiempo de paso por el perihelio
utc_time2 = "2006-06-30 00:00:00"

days_diff = days_between(utc_time1, utc_time2)
print(f"Hay Δt = {days_diff:.4f} días entre {utc_time1} y {utc_time2}.")

Hay Δt = 421.0000 días entre 2005-05-05 00:00:00 y 2006-06-30 00:00:00.


In [5]:
M = n*(days_diff)
print('La anomalía media es M = ', M, '°')

La anomalía media es M =  94.65546764523036 °


In [6]:
from scipy.optimize import fsolve 

# Ecuación de Kepler en grados
f = lambda E : M + e*(180/np.pi)*np.sin(np.radians(E)) - E

roots = fsolve(f, [2, 2])
E = roots[0]

print("La anomalía eccéntrica es E = ", E, "°")

La anomalía eccéntrica es E =  108.47699229744289 °


In [7]:
θ = 2*np.arctan(np.sqrt((1+e)/(1-e))*np.tan(np.radians(E/2)))
θ = np.degrees(θ)
print("La anomalía verdadera es θ = ", θ, "°")


La anomalía verdadera es θ =  121.91447829077109 °


In [8]:
r = a*(1-e*np.cos(np.radians(E)))
print("El radio vector del asteroide es r =", r, "U.A.")

El radio vector del asteroide es r = 2.894419414380374 U.A.


## Summary:
* El periodo orbital es T =  1601.1753337699245 días
* El movimiento medio es n =  0.22483484001242365 °/d
* Hay Δt = 421.0000 días entre 2005-05-05 00:00:00 y 2006-06-30 00:00:00.
* La anomalía media es M =  94.65546764523036 °
* La anomalía verdadera es θ =  121.91447829077109 °
* El radio vector del asteroide es r = 2.894419414380374 U.A.